stan/math/rev/core/arena_matrix.hpp

```diff
@@ -225,8 +224,10 @@ class arena_matrix<MatrixType, require_eigen_sparse_base_t<MatrixType>>
    */
   arena_matrix(const arena_matrix<MatrixType>& other)
       : Base::Map(other.rows(), other.cols(), other.nonZeros(),
-                  other.outerIndexPtr(), other.innerIndexPtr(),
-                  other.valuePtr(), other.innernonZeroPtr()) {}
+                  const_cast<StorageIndex*>(other.outerIndexPtr()),
+                  const_cast<StorageIndex*>(other.innerIndexPtr()),
+                  const_cast<Scalar*>(other.valuePtr()),
+                  const_cast<StorageIndex*>(other.innerNonZeroPtr())) {}
```

Comment:
I think I need you to walk me through these const casts. Are the const copy, move, and copy constructors all identical now? That seems odd to me

Reply:
The actual structure just holds pointers to a data that sits on our stack allocator so anytime we copy or "move" we would just be copying pointers. So that makes them all the same.

We don't null out the move constructors input argument, but that's not a necessary condition for a move constructor and I don't really see a reason to do it here